File: backend/SupportTicketManagementSystem.API/Models/Ticket.cs
using SupportTicketManagementSystem.API.Models.Enums;

namespace SupportTicketManagementSystem.API.Models;

public class Ticket
{
    public int Id { get; set; }
    public string TicketNumber { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public TicketPriority Priority { get; set; }
    public TicketStatus Status { get; set; } = TicketStatus.Open;
    public string? AssignedTo { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedDate { get; set; }
    public DateTime? DueDate { get; set; }
    public DateTime? ResolvedDate { get; set; }

    public ICollection<TicketComment> Comments { get; set; } = new List<TicketComment>();
    public ICollection<TicketStatusHistory> StatusHistory { get; set; } = new List<TicketStatusHistory>();
}

File: frontend/src/types/ticket.ts
export type TicketPriority = 'Low' | 'Medium' | 'High' | 'Critical';
export type TicketStatus = 'Open' | 'InProgress' | 'Resolved' | 'Closed';

export interface Ticket {
  id: number;
  ticketNumber: string;
  title: string;
  description: string;
  priority: TicketPriority;
  priorityName: string;
  status: TicketStatus;
  statusName: string;
  assignedTo: string | null;
  createdDate: string;
  updatedDate: string | null;
  dueDate: string | null;
  resolvedDate: string | null;
  commentCount: number;
}

export interface CreateTicketInput {
  title: string;
  description: string;
  priority: TicketPriority;
  status?: TicketStatus;
  assignedTo?: string;
  dueDate?: string;
}

export interface UpdateTicketInput {
  title: string;
  description: string;
  priority: TicketPriority;
  assignedTo?: string;
  dueDate?: string;
}

export interface ChangeStatusInput {
  newStatus: TicketStatus;
  changedBy?: string;
}

export interface TicketQueryParams {
  search?: string;
  status?: string;
  priority?: string;
  assignedTo?: string;
  fromDate?: string;
  toDate?: string;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
}

export interface PagedResult<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}
